dali/pipeline/operators/reader/loader/file_loader.h

```diff
@@ -34,87 +34,36 @@ namespace dali {
 
 namespace filesystem {
 
-void assemble_file_list(const std::string& path, int label,
-                        std::vector<std::pair<std::string, int>> *file_label_pairs) {
-  DIR *dir = opendir(path.c_str());
-  struct dirent *entry;
-
-  const std::vector<std::string> valid_extensions({".jpg", ".jpeg", ".png", ".bmp"});
-
-  while ((entry = readdir(dir))) {
-    std::string full_path = path + "/" + std::string{entry->d_name};
-    struct stat s;
-    stat(full_path.c_str(), &s);
-    if (S_ISREG(s.st_mode)) {
-      std::string full_path_lowercase = full_path;
-      std::transform(full_path_lowercase.begin(), full_path_lowercase.end(),
-                     full_path_lowercase.begin(), ::tolower);
-      for (const std::string& s : valid_extensions) {
-        size_t pos = full_path_lowercase.rfind(s);
-        if (pos != std::string::npos && pos + s.size() == full_path_lowercase.size()) {
-          file_label_pairs->push_back(std::make_pair(full_path, label));
-          break;
-        }
-      }
-    }
-  }
-  closedir(dir);
-}
-
-vector<std::pair<string, int>> traverse_directories(const std::string& path) {
-  // open the root
-  DIR *dir = opendir(path.c_str());
-
-  DALI_ENFORCE(dir != nullptr,
-      "Directory " + path + " could not be opened.");
-
-  struct dirent *entry;
-  int dir_count = 0;
-
-  std::vector<std::pair<std::string, int>> file_label_pairs;
-
-  while ((entry = readdir(dir))) {
-    struct stat s;
-    std::string full_path = path + "/" + std::string(entry->d_name);
-    int ret = stat(full_path.c_str(), &s);
-    DALI_ENFORCE(ret == 0,
-        "Could not access " + full_path + " during directory traversal.");
-    if (strcmp(entry->d_name, ".") == 0 || strcmp(entry->d_name, "..") == 0) continue;
-    if (S_ISDIR(s.st_mode)) {
-      assemble_file_list(full_path, dir_count, &file_label_pairs);
-      dir_count++;
-    }
-  }
-  printf("read %lu files from %d directories\n", file_label_pairs.size(), dir_count);
-
-  closedir(dir);
-
-  return file_label_pairs;
-}
+vector<std::pair<string, int>> traverse_directories(const std::string& path);
 
 }  // namespace filesystem
 
 class FileLoader : public Loader<CPUBackend> {
  public:
-  explicit FileLoader(const OpSpec& spec)
+  explicit inline FileLoader(const OpSpec& spec,
+      vector<std::pair<string, int>> image_label_pairs = std::vector<std::pair<string, int>>())
```

Comment:
I'm just guessing, but this looks like an argument, which would gain from being either `&` or `const &`